### DISCUSSION TITLE
How to create a bucket with region us-east-1

### DISCUSSION THREAD

**Discussion by hantsy** [aws/aws-sdk-kotlin#632] [Q&A] [answered]
I tried to use this SDK to create a bucket with region us-east-1(we use this region in our project).

```kotlin 
val request = CreateBucketRequest {
    bucket = bucketName
    createBucketConfiguration {
        locationConstraint = BucketLocationConstraint.fromValue(properties.region)
    }
}
try {
    client.createBucket(request)
} catch (e: Exception) {
    log.error("Failed to create bucket $bucketName", e)
    throw StorageServiceException(e.message ?: "Failed to create bucket $bucketName")
}
```

It failed, due to there is no `us-east-1` item in the `BucketLocationConstraint`.  

How to fix this?

**Comment by aajtodd** [comment, marked as answer]
From the API reference docs for S3 [CreateBucket](https://docs.aws.amazon.com/AmazonS3/latest/API/API_CreateBucket.html#AmazonS3-CreateBucket-request-LocationConstraint)

> Specifies the Region where the bucket will be created. **If you don't specify a Region, the bucket is created in the US East (N. Virginia) Region (us-east-1).**

Try leaving the location constraint off when creating a bucket in `us-east-1`.

**Reply by hantsy** [reply]
Thanks.
